storage/oasis/client.go

```diff
@@ -22,14 +22,23 @@ type ClientFactory struct {
 }
 
 // NewClientFactory creates a new oasis-node client factory.
-func NewClientFactory(ctx context.Context, network *config.Network) (*ClientFactory, error) {
-	connection, err := connection.Connect(ctx, network)
```

Comment:
`Connect` (as opposed to `ConnectNoVerify`) fetches the chain context from the node and compares it with the one specified (manually, via config) in `network`. The fetching is "not needed" because we specify the chain context in the config. But there is a downside to not checking: If the chain context specified in the config does not match the chain that's actually provided by the node in the config, that will result in errors way later, in the form of violated FK constraints. Not sure why/how that even happens; it probably has to do with the chain context being used to [check the signatures](https://github.com/oasisprotocol/oasis-indexer/blob/f9d135cb05cbf08e682859318a02770f3c9bc51e/storage/oasis/client.go#L54-L54) of some of the node payloads.

Alternatively, we could always fetch the context automatically. But that also increases the danger of accidentally connecting to the wrong node and polluting the DB with completely unrelated information. I was on the fence for what to do here, and eventually went with the safer solution.

Reply:
Do we need to check again that chain context from the node matches our config?